include/libsemigroups/fpsemi-examples.hpp

```diff
@@ -0,0 +1,539 @@
+    return static_cast<author>(static_cast<uint64_t>(auth1)
+                               + static_cast<uint64_t>(auth2));
+  }
+
+  std::vector<relation_type> rook_monoid(size_t l, int q);
+
+  // Construct a presentation for the Renner common type B monoid.
+  //
+  // Returns a vector of relations defining the Renner common type B monoid with
+  // parameters `l` and `q`.
+  //
+  // \param l the degree
+  // \param q the parameter `q`
+  //
+  // \returns A `std::vector<relation_type>`
+  //
+  // \noexcept
+  std::vector<relation_type> renner_common_type_B_monoid(size_t l, int q);
+
+  std::vector<relation_type> RennerTypeBMonoid(size_t l, int q);
+
+  // Construct a presentation for the Renner type B monoid.
+  //
+  // Returns a vector of relations defining the Renner type B monoid with
+  // parameters `l` and `q`.
+  //
+  // The argument `val` determines the specific presentation
+  // which is returned. The options are:
+  // * `author::Godelle` (described
+  // [here](https://doi.org/10.48550/arXiv.0904.0926), and is possibly
+  // incorrect)
+
+  // \param l the parameter `l`
+  // \param q the parameter `q`
+  // \param val the author value
+  //
+  // \returns A `std::vector<relation_type>`
+  //
+  // \throws LibsemigroupsException if the author value is not
+  // `author::Godelle`.
+  std::vector<relation_type> renner_type_B_monoid(size_t l, int q, author val);
+
+  // Construct a presentation for the Renner common type D monoid.
+  //
+  // Returns a vector of relations defining the Renner common type D monoid with
+  // parameters `l` and `q`.
+  //
+  // \param l the parameter `l`
+  // \param q the parameter `q`
+  //
+  // \returns A `std::vector<relation_type>`
+  //
+  // \noexcept
+  std::vector<relation_type> renner_common_type_D_monoid(size_t l, int q);
+
+  // Construct a presentation for the Renner type D monoid.
+  //
+  // Returns a vector of relations defining the Renner type D monoid with
+  // parameters `l` and `q`.
+  //
+  // The argument `val` determines the specific presentation
+  // which is returned. The options are:
+  // * `author::Godelle` (described
+  // in [10.48550/arXiv.0904.0926](https://doi.org/10.48550/arXiv.0904.0926),
+  // and is possibly incorrect)
+
+  // \param l the parameter `l`
+  // \param q the parameter `q`
+  // \param val the author value
+  //
+  // \returns A `std::vector<relation_type>`
+  //
+  // \throws LibsemigroupsException if the author value is not
+  // `author::Godelle`.
+  std::vector<relation_type> renner_type_D_monoid(size_t l, int q, author val);
+
+  std::vector<relation_type> RennerTypeDMonoid(size_t l, int q);
+
+  //! Construct a presentation for the stellar monoid.
+  //!
+  //! Returns a vector of relations giving a semigroup presentation defining the
+  //! stellar monoid with `n` generators, as in Theorem 4.39 of
+  //! [10.48550/arXiv.1910.11740](https://doi.org/10.48550/arXiv.1910.11740).
+  //!
+  //! \param n the number of generators
```

Comment:
```suggestion
  //! \param l the number of generators
```
Should use the same names for parameters as in the actual function signature.